src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -104,93 +112,45 @@ public override async Task<IEnumerable<string>> DownloadVersion(PackageURL purl,
 
             try
             {
-                HttpClient httpClient = CreateHttpClient();
-                JsonDocument doc = await GetJsonCache(httpClient, $"{RegistrationEndpoint}{packageName.ToLowerInvariant()}/index.json");
-                List<string> versionList = new();
-                foreach (JsonElement catalogPage in doc.RootElement.GetProperty("items").EnumerateArray())
+                CancellationToken cancellationToken = CancellationToken.None;
+
+                FindPackageByIdResource resource = await _sourceRepository.GetResourceAsync<FindPackageByIdResource>();
+
+                PackageIdentity packageIdentity = new(purl.Name, NuGetVersion.Parse(purl.Version));
+
+                IPackageDownloader? downloader = await resource.GetPackageDownloaderAsync(
+                    packageIdentity,
+                    _sourceCacheContext,
+                    NullLogger.Instance, 
```

Comment:
I don't believe that there is a mechanism to pass a logger, but I think it would be fine to add an ILoggerFactory parameter and convert the library to use ILogger. (in a separate PR for changing this across the project).

Reply:
Yeah, I might file an issue asking if they can just migrate over to the official implementation now. Any thoughts on that?

Reply:
The NuGet.Protocol library actually defines their own ILogger, it isn't from Microsoft.Extensions.Logging from what I can tell. https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/Logging/ILogger.cs

Reply:
I just attempted to make this wrapper but it is not possible because the ILogger they does not have public members.

Reply:
Added a wrapper @gfs.